src/ImageCaptureCore/ICCameraDevice.cs

```diff
@@ -0,0 +1,92 @@
+using System;
+using Foundation;
+using ObjCRuntime;
+#nullable enable
+
+namespace ImageCaptureCore {
+	partial class ICCameraDevice {
+
+		public delegate void DidReadDataDelegate (NSData data, ICCameraFile file, NSError error);
+
+		public void RequestReadDataFromFile (ICCameraFile file, long offset, long length, DidReadDataDelegate callback)
+		{
+			var actionObject = new DidReadDataFromFileAction (callback);
+			var sel = Selector.FromHandle (Selector.GetHandle (DidReadDataFromFileAction.CallbackSelector));
+			if (sel is not null)
+				RequestReadDataFromFile (file, offset, length, actionObject, sel, IntPtr.Zero);
```

Comment:
This can be simplified, you don't need to create a the handle for a selector, and then get the string for that selector, you can just pass the string directly:

```suggestion
			RequestReadDataFromFile (file, offset, length, actionObject, DidReadDataFromFileAction.CallbackSelector, IntPtr.Zero);
```

Reply:
agreed this could be simpler, but this fix doesn't work as the method expects a selector argument and `DidReadDataFromFileAction.CallbackSelector` is a string. Implemented a workaround by passing the value to a selector constructor instead.